File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/ProtocolPluginFacade.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/**
 * @hidden Public because it's used by IceBT and the 'Ice/background' test.
 */
public final class ProtocolPluginFacade {
    private final Instance _instance;

    /** Registers a plug-in facade. */
    public ProtocolPluginFacade(Communicator communicator) {
        _instance = communicator.getInstance();
    }

    /** Registers an {@link EndpointFactory}. */
    public void addEndpointFactory(EndpointFactory factory) {
        _instance.endpointFactoryManager().add(factory);
    }

    /** Gets an {@link EndpointFactory}. */
    public EndpointFactory getEndpointFactory(short type) {
        return _instance.endpointFactoryManager().get(type);
    }
}

File: csharp/src/Ice/Internal/ProtocolPluginFacade.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.Internal;

/// <summary>Provides limited access to the internal endpoint factory manager. Used by transport plug-ins.</summary>
public sealed class ProtocolPluginFacade
{
    private readonly Instance _instance;

    public ProtocolPluginFacade(Communicator communicator) => _instance = communicator.instance;

    // Registers an EndpointFactory.
    public void addEndpointFactory(EndpointFactory factory) => _instance.endpointFactoryManager().add(factory);

    // Gets an EndpointFactory.
    public EndpointFactory getEndpointFactory(short type) => _instance.endpointFactoryManager().get(type);
}
